### ISSUE TITLE
CodeLens report 0 references for attributes not using constructor

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#46089]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1114799/codelens-report-0-references-for-attributes-not-us.html)._

---
I have a few custom attributes defined in some of my projects. It looks like if they have an explicit constructor with some parameter, CodeLens correctly shows the references over the constructor as expected. But, if the attributes have no constructor and are just added with the standard `[SomeAttribute]` syntax, then I only see "0 references" over the attribute type definition, with no easy way to find the actual usages of that attribute in the project.

CodeLens should correctly report all the actual usages of the attribute in the current project, regardless of how they're defined and added to the respective targets in the codebase.

---
#### Feedback Bot on 7/15/2020, 06:51 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by RikkiGibson** [comment]
Knowing relatively little about how CodeLens works, it sounds like synthesized symbols which share a common location should have their usage counts summed up at that location.

**Comment by CyrusNajmabadi** [comment]
Yeah, i imagine this isn't about attributes per-se, but more about implicit constructors and how references to them are surfaced.

**Comment by sharwell** [comment]
Design review conclusion: Attribute usage should be treated as a use of both the constructor and the type, in the same way that object creation is treated as a use of both the constructor and the type. We would accept a pull request to fix/unify this behavior, but would be especially looking for comprehensive tests demonstrating that inline rename functionality for attributes is not negatively impacted by the change.
